modules/metastore/tests/src/Unit/DatasetApiDocsTest.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace Drupal\Tests\metastore\Unit;
+
+use Drupal\Core\Site\Settings;
+use Drupal\common\DkanApiDocsGenerator;
+use Drupal\common\Plugin\DkanApiDocsBase;
+use Drupal\common\Plugin\DkanApiDocsPluginManager;
+use Drupal\metastore\DatasetApiDocs;
+use Drupal\metastore\SchemaRetriever;
+use Drupal\metastore\Service;
+use Drupal\metastore\Storage\DataFactory;
+use Drupal\metastore\ValidMetadataFactory;
+use MockChain\Chain;
+use PHPUnit\Framework\TestCase;
+use RootedData\RootedJsonData;
+
+/**
+ * Unit tests for DatasetApiDocs.
+ */
+class DatasetApiDocsTest extends TestCase {
+
+  /**
+   *
+   */
+  public function testBuildDatasetApiDocsWithDkanApiBase() {
+
+    $dkanApiPath = '/foo/bar';
+
+    $settings = new Settings(['dkan_api_base' => $dkanApiPath]);
+
+    $generator = new DkanApiDocsGenerator(
+      $this->getManagerChain()->getMock(),
+      $settings
+    );
+
+    $service = $this->getServiceChain()->getMock();
+
+    $datasetDoc = new DatasetApiDocs($generator, $service, $settings);
+
+    $spec = $datasetDoc->getDatasetSpecific('123');
+
+    $this->assertTrue(is_array($spec['paths']));
+
+    $paths = array_keys($spec['paths']);
+    $expected_paths = [
+      $dkanApiPath . '/api/1/metastore/schemas/dataset/items/123',
```

Comment:
Sorry maybe I'm missing something... if I understand the PR correctly wouldn't we want to see /foo/bar/api/1... as the path here?

Reply:
oh right [face-palm emoji]